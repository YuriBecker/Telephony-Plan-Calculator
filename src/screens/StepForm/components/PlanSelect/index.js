import React from 'react';
import { Button, Layout, Text } from '@ui-kitten/components';
import PropTypes from 'prop-types';
import telephonyPlans from '../../../../constants/telephonyPlans';
import colors from '../../../../styles/colors';

const index = ({ selectedPlan, setSelectedPlan }) => {
  return (
    <>
      <Text
        category="h5"
        style={{
          textAlign: 'center',
          marginVertical: 15,
          color: colors.primary,
        }}
      >
        Qual o plano escolhido?
      </Text>
      <Layout
        style={{
          flexDirection: 'column',
          justifyContent: 'center',
          margin: 15,
          backgroundColor: colors.black[1],
        }}
      >
        {telephonyPlans.map(plan => (
          <Button
            previousBtnText="Voltar"
            onPress={() => setSelectedPlan(plan.text)}
            key={plan.text}
            style={{ marginVertical: 15 }}
            size="large"
            appearance={plan.text === selectedPlan ? 'outline' : 'filled'}
          >
            {plan.text}
          </Button>
        ))}
      </Layout>
    </>
  );
};

export default index;

index.propTypes = {
  selectedPlan: PropTypes.string,
  setSelectedPlan: PropTypes.func.isRequired,
};

index.defaultProps = {
  selectedPlan: null,
};
