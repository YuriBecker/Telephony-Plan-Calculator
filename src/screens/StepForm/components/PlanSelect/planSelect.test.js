import React from 'react';
import { shallow } from 'enzyme';

import PlanSelect from './index';

describe('Testing PlanSelect component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PlanSelect selectedPlan="" setSelectedPlan={jest.fn()} />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render the  Button', () => {
    expect(wrapper.find('ButtonComponent')).toExist();
  });

  it('should render the title Text', () => {
    expect(wrapper.find('TextComponent')).toExist();
  });

  it('should render the  Layout container', () => {
    expect(wrapper.find('LayoutComponent')).toExist();
  });
});
