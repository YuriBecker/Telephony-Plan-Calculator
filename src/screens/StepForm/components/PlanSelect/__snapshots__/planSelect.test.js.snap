// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PlanSelect component should render correctly 1`] = `
<Fragment>
  <TextComponent
    category="h5"
    style={
      Object {
        "color": "#1db954",
        "marginVertical": 15,
        "textAlign": "center",
      }
    }
  >
    Qual o plano escolhido?
  </TextComponent>
  <LayoutComponent
    style={
      Object {
        "backgroundColor": "#212121",
        "flexDirection": "column",
        "justifyContent": "center",
        "margin": 15,
      }
    }
  >
    <ButtonComponent
      appearance="filled"
      key="FaleMais 30"
      onPress={[Function]}
      previousBtnText="Voltar"
      size="giant"
      style={
        Object {
          "marginVertical": 15,
        }
      }
    >
      FaleMais 30
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      key="FaleMais 60"
      onPress={[Function]}
      previousBtnText="Voltar"
      size="giant"
      style={
        Object {
          "marginVertical": 15,
        }
      }
    >
      FaleMais 60
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      key="FaleMais 120"
      onPress={[Function]}
      previousBtnText="Voltar"
      size="giant"
      style={
        Object {
          "marginVertical": 15,
        }
      }
    >
      FaleMais 120
    </ButtonComponent>
  </LayoutComponent>
</Fragment>
`;
